compiler/util/files.cpp

```diff
@@ -373,15 +376,20 @@ void testInputFiles(int numFilenames, char* filename[]) {
       closeInputFile(testfile);
     }
 
-    inputFilenames[i] = astr(filename[i]);
+    inputFilenames[cursor] = astr(filename[i]);
   }
 
-  inputFilenames[i] = NULL;
+  inputFilenames[cursor] = NULL;
 
   if (!foundChplSource && fUseIPE == false)
     USR_FATAL("Command line contains no .chpl source files");
 }
 
+void testInputFile(const char* filename) {
```

Comment:
This isn't your fault since you're following `testInputFiles` but this name isn't intuitive to me. `addAndTestInputFile`,  `addAndCheckSourceFile`, or maybe just `addSourceFile` (with a comment that says it will check if it can open it) seem better to me.

I think this name should stay in sync with `testInputFiles` though (i.e. I think it's fine for you to let this name stay as it is for now, or if it seems worthwhile to change it, I think `testInputFiles` and `testInputFile` should both be changed.)